activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1148,7 +1148,26 @@ def type_to_sql(type, limit: nil, precision: nil, scale: nil, **) # :nodoc:
       #   columns_for_distinct("posts.id", ["posts.created_at desc"])
       #
       def columns_for_distinct(columns, orders) # :nodoc:
-        columns
+        order_columns = orders.compact_blank.map { |s|
```

Comment:
I'm curious why returning order_columns for all adapters?
AFAIK this is not necessary for SQLite3.

Reply:
AFAIK, this is ANSI SQL standard.

I undid for SQLite3 to be simpler.

Reply:
AFAIK, this is ANSI SQL standard.

I undid for SQLite3 to be simpler.

Reply:
@kamipo I have fixed to return simple query for SQLite3

Since this is typical behavior for ANSI SQL standard, I thought it better to make default behavior rather than override for mysql, postgres (and probably others)

Does this change fix your issues?

Reply:
@kamipo are you ok with this pr now?

anything else that would help you understand / merge?